src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -96,13 +97,27 @@ static ClientApplicationBase()
         /// Gets or sets correlation Id which would be sent to the service with the next request.
         /// Correlation Id is to be used for diagnostics purposes.
         /// </summary>
-        public Guid CorrelationId { get; set; }
-
+        private Guid _correlationId;
+        public Guid CorrelationId
+        {
+            get
+            {
+                return _correlationId;
+            }
+            set
+            {
+                _correlationId = value;
+                RequestContext = this.CreateRequestContext(value);
+            }
+        }
+        
         /// <summary>
         /// Gets a value indicating whether address validation is ON or OFF.
         /// </summary>
         public bool ValidateAuthority { internal get; set; }
 
+        private RequestContext RequestContext { get; set; }= new RequestContext(Guid.Empty);
```

Comment:
insert space before =. I understand that resharper is turned off. We can always fix the formatting in next iteration.